### DISCUSSION TITLE
Unable to use pg_net within a cron execution

### DISCUSSION THREAD

**Discussion by mcrrobinson** [supabase/supabase#29436] [Questions]
Running this kind of task from the `SQL Editor` works fine.
```sql
SELECT 
  net.http_post(
    url:='https://httpbin.org/post',
    body:='{"hello": "world"}'::jsonb,
    timeout_milliseconds := 5000
  )
```

Creating a cron job function such as
```SQL
CREATE OR REPLACE FUNCTION public.schedule_cron_job (alert_id TEXT, cron_string TEXT) RETURNS VOID AS $$
BEGIN
  -- Schedule the cron job with the provided alert_id and cron_string
  PERFORM
    cron.schedule (
      alert_id,  -- Use the provided alert_id
      cron_string,  -- Use the provided cron_string
      $sql$
      SELECT 
        net.http_post(
          url:='https://httpbin.org/post',
          body:='{"hello": "world"}'::jsonb,
          timeout_milliseconds := 5000
        )
      $sql$
    );
END;
$$ LANGUAGE plpgsql VOLATILE;
```

Results in exclusively `connection failed`
<img width="1113" alt="image" src="https://github.com/user-attachments/assets/ce21b309-473d-48e4-98c3-3fa525df027c">

Tested `pg_net` within the `extensions` schema and the `public` schema both results are identical.

I believe it's a permissions issue however, can't understand what it could be. Any help would be much appreiciated,

Thanks

**Comment by HarshNarayanJha** [comment]
Running into the same issue

**Comment by JulesSantana** [comment]
Same issue, for me it says it detects pg_net not enabled but in fact it is enabled in the public schema in supabase
